subsys/spm/spm.c

```diff
@@ -472,33 +467,29 @@ static void spm_config_peripherals(void)
 
 	};
 
-	if (IS_ENABLED(CONFIG_SPM_NRF_DPPIC_NS)) {
-		spm_dppi_configure(CONFIG_SPM_NRF_DPPIC_PERM_MASK);
-	}
+#if defined(CONFIG_NRF_DPPI_CHANNEL_MASK_SECURE)
+	NRF_SPU->DPPI[0].PERM = CONFIG_NRF_DPPI_CHANNEL_MASK_SECURE;
+#endif
 
 	PRINT("Peripheral\t\tDomain\t\tStatus\n");
 
-	if (IS_ENABLED(CONFIG_SPM_NRF_P0_NS)) {
-		/* Configure GPIO pins to be Non-Secure */
-		NRF_SPU->GPIOPORT[0].PERM = 0;
-	}
+#if defined(CONFIG_NRF_GPIO0_PIN_MASK_SECURE)
+	NRF_SPU->GPIOPORT[0].PERM = CONFIG_NRF_GPIO0_PIN_MASK_SECURE;
+#endif
 
-#ifdef CONFIG_SOC_NRF5340_CPUAPP
-	if (IS_ENABLED(CONFIG_SPM_NRF_P1_NS)) {
-		/* Configure GPIO pins to be Non-Secure */
-		NRF_SPU->GPIOPORT[1].PERM = 0;
-	}
+#if defined(CONFIG_NRF_GPIO1_PIN_MASK_SECURE)
+	NRF_SPU->GPIOPORT[1].PERM = CONFIG_NRF_GPIO1_PIN_MASK_SECURE;
 #endif
 
 	for (size_t i = 0; i < ARRAY_SIZE(periph); i++) {
 		int err;
 
 #ifndef CONFIG_SPM_BOOT_SILENTLY
 		PRINT("%02u %-21s%s", i, periph[i].name,
-		      periph[i].nonsecure ? "Non-Secure" : "Secure\t");
+		      periph[i].secure ? "Secure\t" : "Non-Secure");
```

Comment:
If you check the inverse you will save testers time rewriting their regexes used for testing.

`!secure ? "Non-secure": "Secure"`

Reply:
Unless I'm missing something the output is not affected by this change. the `\t` is in the same place.

Reply:
Ah, I read it incorrectly.